k8s/pkg/apis/longhorn/v1beta2/engineimage.go

```diff
@@ -5,10 +5,9 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 type EngineImageState string
 
 const (
-	EngineImageStateDeploying    = EngineImageState("deploying")
-	EngineImageStateDeployed     = EngineImageState("deployed")
-	EngineImageStateIncompatible = EngineImageState("incompatible")
-	EngineImageStateError        = EngineImageState("error")
```

Comment:
Let's don't remove it, but mark it as deprecated. 

Also, make an upgrade path for updating the `incompatible` field by `status::state::EngineImageStateIncompatible`.